### ISSUE TITLE
v1.13.0 Error: Invalid function argument jsondecode

### ISSUE THREAD

**Issue by adeturner** [Azure/terraform-provider-azapi#464]
Just to say... pushing [breaking changes](https://github.com/Azure/terraform-provider-azapi/releases/tag/v1.13.0) in a point release isn't cricket

```
Error: Invalid function argument

  on aca_app_consumption-ingress-app-dns.tf line 18, in resource "azurerm_dns_txt_record" "dns_txt_rec":
  18:     value = jsondecode(data.azapi_resource.aca_consumption_env_data.output).properties.customDomainConfiguration.customDomainVerificationId
    ├────────────────
    │ while calling jsondecode(str)
    │ data.azapi_resource.aca_consumption_env_data.output is object with 1 attribute "properties"
```

**Comment by ms-henglu** [comment]
Hi @adeturner ,

Thank you for taking time to report this issue and apologize for the trouble.

In the `v1.13.0` release, it introduces the dynamic properties feature, which will make the `output` becomes an HCL object instead of `json` string.

Please remove the `jsondecode` function like 
```
alue = data.azapi_resource.aca_consumption_env_data.output.properties.customDomainConfiguration.customDomainVerificationId
```

**Comment by ms-henglu** [comment]
Hi @adeturner ,

We have released v1.13.1, in this release, a new field called `enable_hcl_output_for_data_source` is introduced, which is used to enable the HCL output for the data source, the default value is `false`.
  This could resolve the following breaking changes in the previous release:
  - `azapi_resource` data source: The `output` field changes from JSON string to HCL object. Users can use access the fields in the output as an HCL object. Please remove the `jsondecode` function when using the `output` field.
  - `azapi_resource_list` data source: The `output` field changes from JSON string to HCL object. Users can use access the fields in the output as an HCL object. Please remove the `jsondecode` function when using the `output` field.
  
  **Please notice if you already upgraded to `v1.13.0` and removed `jsondecode` function for the `output` of data sources, when upgrade to the `v1.13.1` release, you need to set `enable_hcl_output_for_data_source = true` in the `azapi` provider block to enable the HCL output for data sources.**

**Comment by adeturner** [comment]
@ms-henglu thanks, that works great
